fili-core/src/main/java/com/yahoo/bard/webservice/util/AllPagesPagination.java

```diff
@@ -43,7 +43,7 @@ public AllPagesPagination(Collection<T> entireCollection, PaginationParameters p
         this.collectionSize = entireCollection.size();
         this.pageToFetch = paginationParameters.getPage(entireCollection.size());
         this.countPerPage = paginationParameters.getPerPage();
-        this.lastPage = (collectionSize > countPerPage) ? (collectionSize - 1) / countPerPage + 1 : 1;
+        this.lastPage = (collectionSize > countPerPage) ? (int) ((collectionSize - 1) / countPerPage + 1) : 1;
```

Comment:
We should make `lastPage` `long` as well. What happens if someone asks for a lot of dimension rows, with a page size of 1?

Reply:
Reverting to a error if we get greater than maxint since that doesn't seem to be compatible with our scoredoc collector that we're using.